### PR TITLE
Zarr Python 2.3 release post

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR adds a blog post about the Zarr Python 2.3 release.

### CODE REVIEW

**Comment by alimanfoo** [comment]
Hi @zarr-developers/core-devs, this PR brings in draft content for a blog post about the 2.3 Python release, as well as mentioning some future directions including work on new specs. It's WIP but any suggestions or comments very welcome.

File: _posts/2019-05-02-zarr-2.3-release.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+## New storage options for distributed computing
+
+A key feature of the Zarr protocol is that the underlying storage
+system is decoupled from other components via a simple key/value
+interface. In Python, this interface corresponds to the
+[`MutableMapping` interface](@@TODO), which is the interface that
+Python dictionaries (`dict`) implement. The simplicity of this
+interface means it is relatively straightforward to add support for a
+range of different storage systems. The 2.3 release adds support for
+storage using [SQLite](@@TODO), [Redis](@@TODO), [MongoDB](@@TODO) and
```
Comment (jakirkham) [inline _posts/2019-05-02-zarr-2.3-release.md]:
```suggestion
storage using [SQLite]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.SQLiteStore ), [Redis]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.RedisStore ), [MongoDB]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.MongoDBStore ) and
```

File: _posts/2019-05-02-zarr-2.3-release.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+[Azure Blob Storage](@@TODO).
```
Comment (jakirkham) [inline _posts/2019-05-02-zarr-2.3-release.md]:
```suggestion
[Azure Blob Storage]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.ABSStore ).
```

Note: Sorry for the second comment like this. GitHub didn't allow me to consolidate them.

File: _posts/2019-05-02-zarr-2.3-release.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+Here's the same example but storing the data in the cloud via Azure
+Blob Storage:
+
+{% highlight python %}
+TODO
+{% endhighlight %}
+
+Support for other cloud object storage storage services was already
+available via other packages, with Amazon S3 supported via the
+[s3fs](@@TODO) package, and Google Cloud Storage supported via the
```
Comment (jakirkham) [inline _posts/2019-05-02-zarr-2.3-release.md]:
```suggestion
[s3fs]( http://s3fs.readthedocs.io/en/latest/ ) package, and Google Cloud Storage supported via the
```

File: _posts/2019-05-02-zarr-2.3-release.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+[gcsfs](@@TODO) package.
```
Comment (jakirkham) [inline _posts/2019-05-02-zarr-2.3-release.md]:
```suggestion
[gcsfs]( https://gcsfs.readthedocs.io/en/latest/ ) package.
```

File: _posts/2019-05-02-zarr-2.3-release.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+{% highlight python %}
+TODO
+{% endhighlight %}
+
+## Compatibility with N5
+
+Around the same time that development on Zarr was getting started, a
+separate team led by Stephan Saafeld at the Janelia research campus
+was experiencing similar frustrations trying to use HDF5 for storing
+brain imaging data, and developed a software library called
```
Comment (jakirkham) [inline _posts/2019-05-02-zarr-2.3-release.md]: Electron microscopy data? Also IDK how I feel about "brain" as it feels non-technical, but I don't know how much technical verbage we want to include here. @axtimwalde, thoughts?

File: _posts/2019-05-02-zarr-2.3-release.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+If you would like to stay in touch with where things are going, keep
+an eye on the [zarr](@@TODO) and [zarr-specs](@@TODO) GitHub
+repositories, join the [gitter channel](@@TODO), and please feel free
+to raise issues or add comments if you have any questions or ideas.
+
+## Further reading
+
+* [Zarr Python 2.3 release notes](@@TODO)
+* anything else...
```
Comment (jakirkham) [inline _posts/2019-05-02-zarr-2.3-release.md]: Should we engage readers (e.g. "We could really use your help with X?")?

Also do we want to mention that a few of us will be at SciPy?

**Comment by jakirkham** [comment]
Looks pretty good. Sorry I started filling in the links, but then figured you probably want high-level feedback to start. Feel free to dismiss those edit comments if you like. Could use a reaching out section at the end. Also do we want to discuss improvements in Numcodecs?

File: _posts/2019-05-02-zarr-2.3-release.md (hunk 6 above)
Comment (alimanfoo) [reply _posts/2019-05-02-zarr-2.3-release.md]: > Should we engage readers (e.g. "We could really use your help with X?")?

Good idea, there's lots of things we could use help with but didn't want to overwhelm, not sure what to highlight. I figured engagement in the new spec development is the really big thing, could emphasise that a bit more. Happy to add anything else you think.

> Also do we want to mention that a few of us will be at SciPy?

Good idea!

File: _posts/2019-05-02-zarr-2.3-release.md (hunk 5 above)
Comment (alimanfoo) [reply _posts/2019-05-02-zarr-2.3-release.md]: I was hoping you and/or @axtimwalde could improve this :-)

**Comment by alimanfoo** [comment]
> Looks pretty good. Sorry I started filling in the links, but then figured you probably want high-level feedback to start. Feel free to dismiss those edit comments if you like. 

Thanks a lot for those edits, I'm really pressed for time at the moment and any little helps :+1: 

> Could use a reaching out section at the end. 

Sounds good, could you suggest some edits there?

> Also do we want to discuss improvements in Numcodecs?

Would be great to surface work on numcodecs, but maybe as a separate post?

**Comment by jakirkham** [comment]
I'll try to follow up with more edits on other points then.

The main reason I ask about Numcodecs is it might be valuable to mention improved buffer protocol support. Though this is both in Numcodecs and Zarr. So would make sense to mention in this blogpost.

**Comment by alimanfoo** [comment]
> I'll try to follow up with more edits on other points then.

That would be awesome, thanks.

> The main reason I ask about Numcodecs is it might be valuable to mention improved buffer protocol support. Though this is both in Numcodecs and Zarr. So would make sense to mention in this blogpost.

Cool, sounds good, please feel free to add something on that.

**Comment by alimanfoo** [comment]
I've done some editing to fill in all the links and examples, and make it (more-or-less) readable. 

@jakirkham would you like to add something in here about numcodecs work? Happy to hold on to this for a little longer if you can't get to it immediately.

**Comment by alimanfoo** [comment]
@zarr-developers/core-devs any comments? 

@rabernat I've embedded a slide from your cloud/xarray/zarr talk, hope that's OK.

@axtimwalde / @funkey there's a section here about making the connection between zarr and n5, please feel free to edit or expand.

**Comment by jakirkham** [comment]
> @jakirkham would you like to add something in here about numcodecs work? Happy to hold on to this for a little longer if you can't get to it immediately.

Sorry for the delay. Have pushed a short section on buffer protocol support in Zarr and Numcodecs. Please feel free to edit directly as needed.

**Comment by rabernat** [comment]
I think this blog post is excellent and interesting! And In the words of Matt Rocklin, even a lousy blog post is better than no blog post.

I suggest we publish this asap.

**Comment by jakirkham** [comment]
Happy to publish as well.

**Comment by alimanfoo** [comment]
Thanks @jakirkham, awesome. I've added a small bit of context to the section on buffer support, and also added a final section on scipy. Hope that's all OK, going to publish now.

**Comment by alimanfoo** [comment]
[Published](https://zarr-developers.github.io/zarr/python/release/2019/05/23/zarr-2.3-release.html) and [tweeted](https://twitter.com/alimanfoo/status/1139096112139292672) :-)

**Comment by jakirkham** [comment]
Thanks for the edits and getting this out there. 😄

**Pull request merged**